backend/src/main/java/com/redhat/cloud/notifications/events/FromCamelHistoryFiller.java

```diff
@@ -47,13 +65,20 @@ void init() {
         messagesErrorCounter = meterRegistry.counter(MESSAGES_ERROR_COUNTER_NAME);
     }
 
+    @Channel(EGRESS_CHANNEL)
+    Emitter<String> emitter;
+
+    @ConfigProperty(name = "reinject.enabled", defaultValue = "true")
+    boolean enableReInject;
+
     @Acknowledgment(Acknowledgment.Strategy.POST_PROCESSING)
     @Incoming(FROMCAMEL_CHANNEL)
     // Can be modified to use Multi<Message<String>> input also for more concurrency
     public Uni<Void> processAsync(Message<String> input) {
-        return Uni.createFrom().item(() -> input.getPayload())
+        return Uni.createFrom().item(input::getPayload)
                 .onItem().invoke(payload -> log.infof("Processing return from camel: %s", payload))
                 .onItem().transform(this::decodeItem)
+                .onItem().invoke(this::reinjectIfNeeded)
```

Comment:
```suggestion
                .call(() -> reinjectIfNeeded())
```

Reply:
Wouldn't work because `reinjectIfNeeded()` is returning `void`